tests/jdom/adapters.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { suite, test, afterEach } from "mocha"
+import { JDBus } from "../../src/jdom/bus"
+import JDServiceProvider from "../../src/jdom/serviceprovider"
+import { EVENT, SRV_BUTTON, SRV_BUTTON_GESTURE, DEVICE_ANNOUNCE } from "../../src/jdom/constants";
+import { mkBus } from "../testutils";
+
+import ButtonGestureAdapter from "../../src/servers/buttongestureadapter"
+import ButtonServer from "../../src/servers/buttonserver"
+import { JDEvent } from "../../src/jdom/event";
+import { JDDevice } from "../../src/jdom/device";
+
+suite('bus', () => {
+    let bus: JDBus;
+
+    afterEach(() => bus?.stop())
+
+    test('self announce', function(done) {
+        console.log("start test")
+        bus = mkBus();
+        console.log("bus created")
+
+        const options = {  // no idea what this does, cargo-culting from servers.ts
+            resetIn: false,
+        }
+        const buttonServer = new ButtonServer("B0")  // TODO does the name do anything?
+        bus.addServiceProvider(new JDServiceProvider([buttonServer], options))
+
+        bus.on(DEVICE_ANNOUNCE, (device: JDDevice) => {
+            const servicesNames = device.services().map(service => service.specification.classIdentifier)
+
+            // When the button is connected, setup the button adapter and queue test stimulus
+            if (servicesNames.includes(SRV_BUTTON)) {
+                const button = bus.services({serviceClass: SRV_BUTTON})[0]
+                const buttonAdapter = new ButtonGestureAdapter(button, "BG0")
+                bus.addServiceProvider(new JDServiceProvider([buttonAdapter], options))
+
+                // Simple test stimulus, click cycle
+                setTimeout(() => { buttonServer.down() }, 300)
+                setTimeout(() => { buttonServer.up() }, 400)  // should generate click event
```

Comment:
proposed update: 

    await bus.delay(400)
    buttonServer.up()

In general, don't use setTimeout and use async/await.

Reply:
`bus.delay` is something that needs to be implemented, right? (I don't see it in IntelliSense)

Actually, a related question is, should timing functionality go on the bus? On one hand, it's the best place currently for abstracting away the environment. On the other hand, timing isn't a logical function of the bus - and while it could make intuitive sense to do `bus.delay` in test stimulus code, is it also the right place for delays / callbacks in adapters?

Reply:
merge origin/main